### DISCUSSION TITLE
LottieAnimationView's animationLayer not resetted to nil after animation = nil

### DISCUSSION THREAD

**Discussion by ainopara** [airbnb/lottie-ios#2034] [General]
I utilize my custom image provider to verify if there are Lottie JSON files that cannot find corresponding images. Additionally, each Lottie animation requires its unique image provider with a specific search path to find the corresponding image. 

While attempting to reuse a Lottie animation view to display different Lottie animations, I discovered that changing either the image provider or the animation leads to image reloading. Consequently, the image provider and animation do not match, which triggers my custom image missing alert. 

I attempted to resolve this issue by first setting the animation to nil, then changing the image provider, and finally setting the new animation. However, due to LottieAnimationView's animationLayer not being reset to nil after setting animation = nil, the image missing alert is still triggered.

Is it safe to also set animationLayer to nil rather than only remove it from super layer, so that I can avoid the image missing alert triggered unexpectly?

https://github.com/airbnb/lottie-ios/blob/43a8c7ff7109fcc0bde6366e2778d880d8bc3a06/Sources/Public/Animation/LottieAnimationView.swift#L1100-L1112

**Comment by calda** [comment]
That seems reasonable to me @ainopara. Could you open a pull request?

**Reply by ainopara** [reply]
Sure, https://github.com/airbnb/lottie-ios/pull/2035
